honestcash-v2/src/app/main/services/story.service.ts

```diff
@@ -0,0 +1,45 @@
+import {Injectable} from '@angular/core';
+import {forkJoin, Observable} from 'rxjs';
+import {HttpService} from '../../../core';
+
+export const API_ENDPOINTS = {
+  getStory: (id: number) => `/post/${id}`,
+  getStoryUpvotes: (id: number) => `/post/${id}/upvotes`,
+  getStoryUnlocks: (id: number) => `/post/${id}/unlocks`,
+  getStoryComments: (id: number) => `/v2/post/${id}/responses`,
+};
+
+@Injectable({providedIn: 'root'})
+export class StoryService {
+
+  constructor(
+    private http: HttpService,
+  ) {
+  }
+
+  public getStoryWithoutDetails(id: number): Observable<any> {
+    return this.http.get(API_ENDPOINTS.getStory(id));
+  }
+
+  public getStoryWithDetails(id: number): Observable<any> {
```

Comment:
this is not in async manner meaning that forkJoin waits for all observables to complete then returns a result. later we I plan to make it async so that story loads but if there are many comments etc there will be an individual loading indicator in place of comments/unlocks/upvotes (not the entire page) so that they are lazily loaded